test/ExecutionFramework.cpp

```diff
@@ -282,3 +282,21 @@ bool ExecutionFramework::storageEmpty(h160 const& _addr) const
 	}
 	return true;
 }
+
+vector<solidity::frontend::test::LogRecord> ExecutionFramework::recordedLogs() const
+{
+	size_t logIndex{0};
+	vector<solidity::frontend::test::LogRecord> logs;
+	for (evmc::MockedHost::log_record const& logRecord: m_evmcHost->recorded_logs)
+	{
+		solidity::frontend::test::LogRecord record;
+		record.index = logIndex;
+		record.data = bytes{logRecord.data.begin(), logRecord.data.end()};
+		record.creator = EVMHost::convertFromEVMC(logRecord.creator);
+		for (evmc::bytes32 const& topic: logRecord.topics)
+			record.topics.emplace_back(EVMHost::convertFromEVMC(topic));
+		logs.emplace_back(record);
+		++logIndex;
+	}
```

Comment:
```suggestion
		logs.emplace_back(
			logIndex++,
			EVMHost::convertFromEVMC(logRecord.creator),
			bytes{logRecord.data.begin(), logRecord.data.end()},
			logRecord.topics | ranges::views::transform(EVMHost::convertFromEVMC) | ranges::to<vector>,
		);
```

You may need to tweak the argument to `transform()` a bit in the above but it should generally work with pointers to static functions and even pointers to members.

Reply:
I changed it to `logRecord.topics | ranges::views::transform([](evmc::bytes32 _bytes) { return EVMHost::convertFromEVMC(_bytes); }) | ranges::to<vector>`.

Reply:
Right. Out of curiosity I checked why `&EVMHost::convertFromEVMC` won't work here and looks like it's simply because it's overloaded so the compiler needs to infer its type. You could get it to work by casting it to the right type but at that point it starts to look less readable than the lambda :)

```c++
ranges::views::transform(static_cast<h256 (*)(evmc::bytes32 const& _data)>(&EVMHost::convertFromEVMC))
```